recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -312,15 +340,66 @@ class ManagedFirmwareUpdate(FirmwareUpdate):
             while choice > len(firmware_names) or choice < 1:
                 print("Out of range, please reinput your choice:")
                 choice = int(input("-> "))
-            
+
             firmware_name = firmware_names[choice - 1]
 
         print("Will write {} to the SPI flash".format(firmware_name))
 
         self.firmware.seek(0)
-        self.firmware = tarfile.open(fileobj=self.firmware).extractfile(firmware_name)
-        super().update_firmware()
+        with tarfile.open(fileobj=self.firmware) as f:
+            self.firmware = f.extractfile(firmware_name)
+            f.extract(member=self.uboot_default_env_file, path='.')
```

Comment:
Same here, avoid to extract it, but use it via `f.extractfile()` just as the above line.

Reply:
extractfile would return [io.BufferedReader] object

what would we want here is a writing ability.
so for this intent, extract the file and do some operation on this file